src/base/OpenMCCellAverageProblem.C

```diff
@@ -1835,39 +1872,87 @@ OpenMCCellAverageProblem::meshFilter()
   {
     const auto & translation = _mesh_translations[i];
     auto meshFilter = dynamic_cast<openmc::MeshFilter *>(openmc::Filter::create("mesh"));
-    meshFilter->set_mesh(mesh_index);
+    meshFilter->set_mesh(_mesh_index);
     meshFilter->set_translation({translation(0), translation(1), translation(2)});
     mesh_filters.push_back(meshFilter);
   }
 
   return mesh_filters;
 }
 
+void
+OpenMCCellAverageProblem::resetTallies()
+{
+  if (_needs_global_tally)
+  {
+    // erase tally
+    auto idx = openmc::model::tallies.begin() + _global_tally_index;
+    openmc::model::tallies.erase(idx);
+  }
+
+  auto idx = openmc::model::tallies.begin() + _local_tally_index;
+  switch (_tally_type)
+  {
+    case tally::cell:
+    {
+      // erase tally
+      openmc::model::tallies.erase(idx);
+
+      // erase filter
+      auto fidx = openmc::model::tally_filters.begin() + _filter_index;
+      openmc::model::tally_filters.erase(fidx);
+      break;
+    }
+    case tally::mesh:
+    {
+      // erase tallies
+      openmc::model::tallies.erase(idx, idx + _mesh_translations.size());
+
+      // erase filters
+      auto fidx = openmc::model::tally_filters.begin() + _filter_index;
+      openmc::model::tally_filters.erase(fidx, fidx + _mesh_translations.size());
+
+      // erase meshes
+      auto midx = openmc::model::meshes.begin() + _mesh_index;
```

Comment:
I'm not sure we need to erase the meshes. If the `libMesh` mesh updates, the OpenMC mesh should automatically update (i.e. `n_bins`, `get_bin`, and such should all behave correctly.

Reply:
Hrmm though I'm not sure if libMesh point locators will update automatically. Lemme check...

Reply:
In general I think we will need to erase the meshes, because this code is developed for the purpose of either (i) adaptive mesh refinement or (ii) moving meshes. In both cases, the actual mesh is different, so I think we'd just be wasting memory storing an old mesh?

Reply:
The safest thing to do for now is to recreate them though, so we can always come back and remove this extra work later.

Reply:
That may be the case, good point. But at least right now, I think the particular functions being called in OpenMC are not designed for adaptive mesh refinement. For instance,

```
int LibMesh::n_bins() const
{
  return m_->n_elem();
}
```

I'm pretty sure this is going to return the sum of the "active" elements and "inactive" elements for AMR, and not the "actual" elements I currently want to tally on in my mesh.

Reply:
> Will removing & re-adding the mesh help with that though?

No, I guess it wouldn't - sorry, I'm confusing a few different (closely-related) topics we're discussing.

> to ensure the point locators attached to the OpenMC mesh object are updated correctly.

Do you mean that you talked with libMesh devs and they said the point locators wouldn't be updated? Just want to be sure I understand if I should put a TODO comment in Cardinal to remind us that we might remove it in the future.

Reply:
Posting slack convo here for posterity, re: point locators

![Screen Shot 2023-02-27 at 11 49 20 AM](https://user-images.githubusercontent.com/17039662/221667778-ec22fc7b-6b21-4454-b814-153b085f475e.png)

Reply:
It's my hope that we can remove it in the future, so if you're okay with it let's pop a TODO in there or make an issue